tokenizer.go

```diff
@@ -201,6 +224,10 @@ func (t *Tokenizer) EncodeWithOptions(str string, addSpecialTokens bool, opts ..
 		encoding.AttentionMask = uintVecToSlice(res.attention_mask, len)
 	}
 
+	if encOptions.ReturnOffsets && res.offsets != nil {
+		encoding.Offsets = offsetVecToSlice(res.offsets, len)
```

Comment:
i believe you still need to free `res.offsets`

Reply:
I added a line to free the offsets inside the if statement.

Reply:
SGTM